src/uu/split/src/split.rs

```diff
@@ -783,6 +888,26 @@ impl Settings {
                 }
             }
         }
+
+        // Make sure that separator is only one UTF8 character (if specified)
+        let separator = match matches.get_one::<String>(OPT_SEPARATOR) {
+            Some(s) => {
+                if s.chars().count() == 1 {
+                    Some(s.as_bytes()[0])
+                } else {
+                    // handle special cases for '\n' and '\0'
+                    if s == "\\n" {
+                        Some(b'\n')
+                    } else if s == "\\0" {
+                        Some(b'\0')
+                    } else {
+                        return Err(SettingsError::MultiCharacterSeparator(s.to_owned()));
+                    }
+                }
+            }
+            _ => None,
+        };
```

Comment:
can probably be simplified by something like:
```suggestion
let separator = matches.get_one::<String>(OPT_SEPARATOR).and_then(|s| {
    match s.as_str() {
        "\\n" => Some(b'\n'),
        "\\0" => Some(b'\0'),
        _ if s.chars().count() == 1 => Some(s.as_bytes()[0]),
        _ => return Err(SettingsError::MultiCharacterSeparator(s.to_owned())),
    }
});

```

Reply:
fantastic, thank you! will refactor